files/image_config/caclmgrd/caclmgrd

```diff
@@ -347,13 +360,20 @@ class ControlPlaneAclManager(object):
                         # try to do it now. We attempt to determine heuristically based on
                         # whether the src or dst IP of this rule is an IPv4 or IPv6 address.
                         if not table_ip_version:
-                            if (("SRC_IPV6" in rule_props and rule_props["SRC_IPV6"]) or
-                                ("DST_IPV6" in rule_props and rule_props["DST_IPV6"])):
+                            if self.is_rule_ipv6(rule_props):
                                 table_ip_version = 6
-                            elif (("SRC_IP" in rule_props and rule_props["SRC_IP"]) or
-                                  ("DST_IP" in rule_props and rule_props["DST_IP"])):
+                            elif self.is_rule_ipv4(rule_props):
                                 table_ip_version = 4
 
+                        if (self.is_rule_ipv6(rule_props) and (table_ip_version == 4)):
+                            log_error("CtrlPlane ACL table {} is a IPv4 based table and rule {} is a IPV6 rule!"
+                                                          .format(table_name, rule_id))
+                            acl_rules.pop(rule_props["PRIORITY"])
+                        elif (self.is_rule_ipv4(rule_props) and (table_ip_version == 6)):
+                            log_error("CtrlPlane ACL table {} is a IPv6 based table and rule {} is a IPV4 rule!"
+                                                          .format(table_name, rule_id))
+                            acl_rules.pop(rule_props["PRIORITY"])
```

Comment:
@madhanmellanox, @stepanblyschak: Since I asked to remove the exception throwing, this code now simply logs error messages if a table contains a mixture of V4 and V6 rules, but it will continue to process the table. Is this behavior acceptable? One thought I have is that if we find a mixture of rules, we could also set a flag here and skip this table.

Reply:
@jleveque I have added this statement to remove the ACL rule  "acl_rules.pop(rule_props["PRIORITY"])", when a conflicting rule is met in both IPv4 case and IPv6 in my last commit.